src/main/java/de/komoot/photon/query/ReverseRequestFactory.java

```diff
@@ -14,26 +17,90 @@
 public class ReverseRequestFactory {
     private final LanguageChecker languageChecker;
     private final static GeometryFactory geometryFactory = new GeometryFactory(new PrecisionModel(), 4326);
+    
+    protected static HashSet<String> m_hsRequestQueryParams = new HashSet<>(Arrays.asList("lang", "lon", "lat", "radius", "queryStringFilter", "distanceSort", "limit"));
 
     public ReverseRequestFactory(Set<String> supportedLanguages) {
         this.languageChecker = new LanguageChecker(supportedLanguages);
     }
        
     public <R extends ReverseRequest> R create(Request webRequest) throws BadRequestException {
+        
+        
+        for(String queryParam: webRequest.queryParams())
+            if(!m_hsRequestQueryParams.contains(queryParam))
+                throw new BadRequestException(400, "unknown query parameter '" + queryParam + "'.  Allowed parameters are: " + m_hsRequestQueryParams);
+        
+        
         String language = webRequest.queryParams("lang");
         language = language == null ? "en" : language;
         languageChecker.apply(language);
         
-        Point location = null;
+        Point location;
         try {
             Double lon = Double.valueOf(webRequest.queryParams("lon"));
+            if(lon > 180.0 || lon < -180.00){
+                throw new BadRequestException(400, "invalid search term 'lon', expected number >= -180.0 and <= 180.0");
+            }
             Double lat = Double.valueOf(webRequest.queryParams("lat"));
+            if(lat > 90.0 || lat < -90.00){
+                throw new BadRequestException(400, "invalid search term 'lat', expected number >= -90.0 and <= 90.0");
+            }
             location = geometryFactory.createPoint(new Coordinate(lon, lat));
-        } catch (Exception nfe) {
+        } catch (NumberFormatException nfe) {
+            throw new BadRequestException(400, "invalid search term 'lat' and/or 'lon': /?lat=51.5&lon=8.0");
+        } catch (NullPointerException nfe) {
```

Comment:
but...when I look on especially this piece of code between the try{} - indeed, I agree its the better choice. It yields to much clearer code here, and its more robust because many possible nullpointer sources are considered. And the error message is still adequate.